clarity_ext/domain/shared_result_file.py

```diff
@@ -1,16 +1,30 @@
 from clarity_ext.domain.artifact import Artifact
+from clarity_ext.utils import get_and_apply
 
 
 class SharedResultFile(Artifact):
 
-    def __init__(self, api_resource=None, id=None, name=None, artifact_specific_udf_map=None):
+    def __init__(self, api_resource=None, id=None, name=None, artifact_specific_udf_map=None,
+                 **kwargs):
         super(SharedResultFile, self).__init__(
             api_resource=api_resource, id=id, name=name,
             artifact_specific_udf_map=artifact_specific_udf_map)
+        self.has_errors = bool(get_and_apply(kwargs, "has_errors", 0, int))
 
     @staticmethod
     def create_from_rest_resource(api_resource, udf_map=None):
-        shared_result_file_udf_map = udf_map.get("SharedResultFile")
+        shared_result_file_udf_map = udf_map.get("SharedResultFile", dict())
         name = api_resource.name
+        specific_udf_map = udf_map["SharedResultFile"]
```

Comment:
FYI: I have some ideas of removing the requirement of a UDF map and instead just use the underlying `udf_` properties which are now automatically mapped on the object. I'm confident that it will make a lot of things much simpler. It will be in the next milestone though.